packages/angular/plugins/component-testing.ts

```diff
@@ -231,11 +231,20 @@ function normalizeBuildTargetOptions(
     buildOptions.scripts = [];
     buildOptions.stylePreprocessorOptions = { includePaths: [] };
   }
-  const { root, sourceRoot } =
+
+  const config =
     buildContext.projectGraph.nodes[buildContext.projectName]?.data;
+
+  if (!config.sourceRoot) {
+    logger.warn(stripIndents`Unable to find the 'sourceRoot' in the project configuration.
+                 You should set the value in the ${buildContext.projectName} project configuration.
+                 Setting the value to the project root - '${config.root}/src'`);
```

Comment:
I'd maybe replace this with: `attempting with the following value: - config.root/src. (Note: this may fail, setting the correct sourceRoot value in project.json will ensure correctness)`